go-tour/webcrawler.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Fetcher interface {
+	// Fetch returns the body of URL and
+	// a slice of URLs found on that page.
+	Fetch(url string) (body string, urls []string, err error)
+}
+
+type crawlResult struct {
+	depth int
+	urls  []string
+}
+
+func crawlOne(url string, depth int, ch chan crawlResult) {
+	body, urls, err := fetcher.Fetch(url)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
yeah, its ok. Channel is using for sending depth and founded urls. In this case we got an error but we need to send result via channel because such url could be not the last one in slice and we need to decrement depth anyway.